src/server/nyplApiClient/index.ts

```diff
@@ -22,13 +22,16 @@ export class NyplApiClientError extends Error {
   }
 }
 
-const nyplApiClient = async (options = { apiName: "platform" }) => {
-  const { apiName } = options
+const nyplApiClient = async ({
+  apiName = "platform",
+  version = "v0.1",
+} = {}) => {
```

Comment:
Should this be more specific ? this is my first time using this client so I'm not super familiar with its use across the repo

Reply:
What do you mean by specific? Also, can you explain what the `= {}` in the function signature does? 
FWIW this looks fine by me!

Reply:
@dgcohen maybe you can confirm if this looks kosher? the username validation endpoint requires v0.3, so i suggested emma make the version be a param. at the same time, i thought this was a good opportunity to trim away the duplicated discovery/platform urls in the config.

Reply:
I don't know actually how it could be more specific, just generally looking for confirmation that this makes sense from people who wrote/used this before

Reply:
I think it's clear enough when you call the function what version is being used since you pass it as an argument. All tests also continue to pass so it shouldn't break anything.

One possible option to make this clearer (maybe?) is to make this a higher-order function. So it can be something like 
```
const baseNyplApiClient = ({
  version = "v0.1",
} = {}) => {
  return async ({
    apiName = "platform",
  } = {}) => {
      if (CACHE.clients[apiName]) {
        return CACHE.clients[apiName]
      }
    
      const baseUrl =
        appConfig.apiEndpoints[apiName][appEnvironment] + "/" + version
    //...
  })
})

export const nyplApiClientv01 = baseNyplApiClient()
export const nyplApiClientv03 = baseNyplApiClient({ version: "v0.3" })
```

And then call either function like they're being called now such as `const platformClient = await nyplApiClientv03()`. 

Caveat: the syntax might be off and I haven't tested this. If you like this approach, do it in another PR because of its use across the repo. But what do you think, @charmingduchess ?

Reply:
I like the look of that!